content/en/docs/gke/private-clusters.md

```diff
@@ -1,301 +1,235 @@
+  * This likely indicates an issue with access to private GCR; this could be an issue with
 
-          1. Remove the images from the `images` section
-          1. Remove it from the `steps` section
+    * DNS configurations - Check that the `DNSRecordSet` and `DNSManagedZone` CNRM resources are in a ready state
+    * Routes - Ensure any default route to the internet has a larger value for the priority 
+        then any routes to private GCP APIs so that the private routes match first.
 
-1. Create a cloud build job to do the mirroring
+        * If image pull errors show IP addresses not the restricted.googleapis.com VIP then you have
+          an issue with networking
 
-   ```
-    gcloud builds submit --async gs://kubeflow-examples/image-replicate/replicate-context.tar.gz --project <project_id> --config cloudbuild.yaml
-   ```
+    * Firewall rules
 
-1. Update your manifests to use the mirror'd images
+* Access to whitelisted (non Google) sites is blocked
 
-   ```
-   kfctl alpha mirror overwrite -i pipeline.yaml
-   ```
+  * The configuration uses CloudNat to allow selective access to sites
 
-1. Edit file “kustomize/istio-install/base/istio-noauth.yaml”: 
+    * In addition to allowing IAP, this allows sites like GitHub to be whitelisted.
 
-   1. Replace `docker.io/istio/proxy_init:1.16` to `gcr.io/<project_id>/docker.io/istio/proxy_init:1.16`
-   1. Replace `docker.io/istio/proxyv2:1.1.6` to `gcr.io/<project_id>/docker.io/istio/proxyv2:1.1.6`
+  * In order for CloudNat to work you need
 
-## Deploy Kubeflow with Private GKE
+    * A default route to the internet
+    * Firewall rules to allow egress traffic to whitelisted sites
 
-{{% alert title="Coming Soon" color="warning" %}}
-You can follow the issue: [Documentation on how to use Kubeflow with private GKE and VPC service controls](https://github.com/kubeflow/website/issues/1705)
-{{% /alert %}}
+      * These rules need to be higher priority then the deny all firewall egress rules.
+
+### Kubernetes Webhooks are blocked by firewall rules
+
+A common failure mode is that webhooks for custom resources are blocked by default firewall rules.
+As explained in the [GKE Docs](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules) only connections from master to ports 443 and 10250
+are allowed by default. If you have a webhook serving on a different port
+you will need to add an explict ingress firewall rule to allow that port to be accessed.
+
+These errors usually manifest as failures to create custom resources that depend on webhooks. An example
+error is
+
+```
+Error from server (InternalError): error when creating ".build/kubeflow-apps/cert-manager.io_v1alpha2_certificate_admission-webhook-cert.yaml": Internal error occurred: failed calling webhook "webhook.cert-manager.io": the server is currently unable to handle the request
+```
```

Comment:
Line 241: since there is already one troubleshooting section for KF on private GKE:

```diff
- * [Troubleshoot](/docs/gke/troubleshooting-gke) any issues you may
  find.
+ * [Troubleshoot](/docs/gke/troubleshooting-gke) any GKE issues you may
  find.
```